rust/scx_utils/src/topology.rs

```diff
@@ -99,13 +99,13 @@ lazy_static::lazy_static! {
     pub static ref NR_CPUS_POSSIBLE: usize = libbpf_rs::num_possible_cpus().unwrap();
 }
 
-#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+#[derive(Debug, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
```

Comment:
added the extra sorting macros because I ran into [this](https://github.com/rust-lang/rust/issues/113550) when trying to sort by `CoreType`.